packages/expo/cli/start/server/webpack/WebpackBundlerDevServer.ts

```diff
@@ -0,0 +1,351 @@
+      middleware,
+      server,
+    });
+
+    return {
+      messageSocket,
+      eventsSocket,
+      debuggerProxy,
+      logReporter,
+      inspectorProxy,
+    };
+  }
+
+  isTargetingNative(): boolean {
+    // Temporary hack while we implement multi-bundler dev server proxy.
+    return ['ios', 'android'].includes(process.env.EXPO_WEBPACK_PLATFORM || '');
+  }
+
+  isTargetingWeb(): boolean {
+    return true;
+  }
+
+  private async createNativeDevServerMiddleware({
+    port,
+    compiler,
+    options,
+  }: {
+    port: number;
+    compiler: AnyCompiler;
+    options: BundlerStartOptions;
+  }) {
+    if (!this.isTargetingNative()) {
+      return null;
+    }
+
+    const { createDevServerMiddleware } = await import('../middleware/createDevServerMiddleware');
+
+    const nativeMiddleware = createDevServerMiddleware({
+      port,
+      watchFolders: [this.projectRoot],
+    });
+    // Add manifest middleware to the other middleware.
+    // TODO: Move this in to expo/dev-server.
+
+    const middleware = this._getManifestMiddleware(options);
+
+    nativeMiddleware.middleware.use(middleware).use(
+      '/symbolicate',
+      createSymbolicateMiddleware({
+        projectRoot: this.projectRoot,
+        compiler,
+        logger: nativeMiddleware.logger,
+      })
+    );
+    return nativeMiddleware;
+  }
+
+  private async getAvailablePortAsync(options: { defaultPort?: number }): Promise<number> {
+    try {
+      const defaultPort = options?.defaultPort ?? 19006;
+      const port = await choosePortAsync(this.projectRoot, {
+        defaultPort,
+        host: WEB_HOST,
+      });
+      if (!port) {
+        throw new CommandError('NO_PORT_FOUND', `Port ${defaultPort} not available.`);
+      }
+      return port;
+    } catch (error) {
+      throw new CommandError('NO_PORT_FOUND', error.message);
+    }
+  }
+
+  async startAsync(options: BundlerStartOptions): Promise<DevServerInstance> {
+    // Do this first to fail faster.
+    const webpack = importWebpackFromProject(this.projectRoot);
+    const WebpackDevServer = importWebpackDevServerFromProject(this.projectRoot);
+
+    await this.stopAsync();
+
+    const { resetDevServer, https, mode } = options;
+
+    const port = await this.getAvailablePortAsync({
+      defaultPort: options.port,
+    });
+
+    this.urlCreator = new UrlCreator(
+      {
+        scheme: https ? 'https' : 'http',
+        ...options.location,
+      },
+      {
+        port,
+        getTunnelUrl: this.getTunnelUrl.bind(this),
+      }
+    );
+
+    Log.debug('Starting webpack on port: ' + port);
+
+    if (resetDevServer) {
+      await clearWebProjectCacheAsync(this.projectRoot, mode);
+    }
+
+    if (https) {
+      Log.debug('Configuring SSL to enable HTTPS support');
+      await ensureEnvironmentSupportsSSLAsync(this.projectRoot).catch((error) => {
+        Log.error(`Error creating SSL certificates: ${error}`);
+      });
```

Comment:
```suggestion
      Log.debug('Configuring TLS to enable HTTPS support');
      await ensureEnvironmentSupportsTLSAsync(this.projectRoot).catch((error) => {
        Log.error(`Error creating TLS certificates: ${error}`);
      });
```

Reply:
Ok but this generates an SSL certificate according to devcert, not sure what the difference is.

Reply:
They are basically the same. There might be some new extensions that are supported only by TLS but for the most part "SSL certificate" and "TLS certificate" are interchangeable. SSL is deprecated though so better to write TLS instead.